<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
  <head>
    <p><a id="logo" href="http://scala-lang.org/"><img src="/sites/default/files/icons/Scala_Logo2008.png" alt="scala-lang.org" border=0/></a></p>
    <title id="Title">Aladdin - Scala Bugtracking</title>
        <style type="text/css">
        table {
            font-size: 10pt;
            background-color: #ebb85c;
        }
        tr.impair        { background-color: #f0be64 }
        tr.impairContrib { background-color: #f0be64 }

#buraq-content {
	float: none;
	color: black;
	background-color: white;
	font-size: 1.25em;
	margin-left: 9.6em;
}
        </style>
      </head>
  <body><!--DisplayBugsBlock_Start-->




<table WIDTH="100%" cellspacing="4" cellpadding="2">

	<tr><td></td>
		<th>[#487]</th>
		<th><strong>project:</strong> nsc</th>
		<th><strong>priority:</strong> low</th>
		<th><strong>category:</strong> bug</th>
        </tr>

        <tr>
          <td></td>
	  <th align="center">submitter</th>
	  <th align="center">assigned to</th>
	  <th align="center">status</th>
	  <th align="center">date submitted</th>
        </tr>

        <tr>
          <td></td>
	  <td align="center" bgcolor="#FFFFFF">Iulian</td>
	  <td align="center" bgcolor="#FFFFFF">Sean</td>
	  <td align="center" bgcolor="#FFFFFF">_</td>
          <td align="center" bgcolor="#FFFFFF">2005-11-16 15:08:58.0</td>

	</tr>

        <tr>
		<th align="right">subject</th>
		<td colspan="4" bgcolor="#FFFFFF">ArrayIndexOutOfBounds on syntactically incorrect programs</td>
	</tr>
	<tr>	
		<th align="right">code</th>
		<td colspan="4" bgcolor="#FFFFFF">
	<font size="-1"><pre>10-1</pre></font></td>
	</tr>
	<tr>
		<th style="text-align:right;">what happened</th>
		<td colspan="4" style="background-color:#FFFFFF;" ><pre><pre>
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: -1
        at scala.tools.nsc.util.Position$class.column(Position.scala:47)
        at scala.tools.nsc.reporters.AbstractReporter$class.scala$tools$nsc$reporters$AbstractReporter$testAndLo\
g(AbstractReporter.scala:49)
        at scala.tools.nsc.reporters.AbstractReporter$class.error(AbstractReporter.scala:37)
        at scala.tools.nsc.CompilationUnits$CompilationUnit$class.error(CompilationUnits.scala:26)
        at scala.tools.nsc.ast.parser.Parsers$Parser$class.syntaxError(Parsers.scala:104)
        at scala.tools.nsc.ast.parser.Parsers$Parser$class.syntaxError(Parsers.scala:100)
        at scala.tools.nsc.ast.parser.Parsers$Parser$class.topStatSeq(Parsers.scala:1634)
        at scala.tools.nsc.ast.parser.Parsers$Parser$class.compilationUnit(Parsers.scala:1783)
        at scala.tools.nsc.ast.parser.Parsers$Parser$class.parse(Parsers.scala:67)
        at scala.tools.nsc.ast.parser.SyntaxAnalyzer$ParserPhase$class.apply(SyntaxAnalyzer.scala:16)
        at scala.tools.nsc.Global$GlobalPhase$class.applyPhase(Global.scala:170)
        at scala.tools.nsc.Global$GlobalPhase$$anon$7.apply(Global.scala:162)
        at scala.tools.nsc.Global$GlobalPhase$$anon$7.apply(Global.scala:162)
        at scala.Iterator$class.foreach(Iterator.scala:262)
        at scala.tools.nsc.Global$GlobalPhase$class.run(Global.scala:162)
        at scala.tools.nsc.Global$Run$class.compileSources(Global.scala:339)
        at scala.tools.nsc.Global$Run$$anon$20.apply(Global.scala:397)
        at scala.tools.nsc.Global$Run$$anon$20.apply(Global.scala:397)
        at scala.runtime.ScalaRunTime$$anon$0.run(ScalaRunTime.scala:34)
        at scala.runtime.RunTime.tryCatch(RunTime.java:79)
        at scala.runtime.ScalaRunTime$$anon$0.<init>(ScalaRunTime.scala:32)
        at scala.runtime.ScalaRunTime$.Try(ScalaRunTime.scala:30)
        at scala.tools.nsc.Global$Run$class.compile(Global.scala:396)
        at scala.tools.nsc.Main$$anon$8.apply(Main.scala:96)
        at scala.tools.nsc.Main$$anon$8.apply(Main.scala:87)
        at scala.runtime.ScalaRunTime$$anon$0.run(ScalaRunTime.scala:34)
        at scala.runtime.RunTime.tryCatch(RunTime.java:79)
        at scala.runtime.ScalaRunTime$$anon$0.<init>(ScalaRunTime.scala:32)
        at scala.runtime.ScalaRunTime$.Try(ScalaRunTime.scala:30)
        at scala.tools.nsc.Main$.process(Main.scala:87)
        at scala.tools.nsc.Main$.main(Main.scala:108)
        at scala.tools.nsc.Main.main(Main.scala:107)
</pre>
</pre></td>
	</tr>
	<tr>
		<th align="right">what expected</th>
		<td colspan="4" bgcolor="#FFFFFF" ><pre>
pos.scala:1: illegal start of class or object definition
10-1
^
one error found
</pre></td>
	</tr>
	<tr>
	
  <td colspan="4" align="right">
      
      <a href="displayBugs.do.html">
        [back to overview]</a></td></tr>

</table>

  <tr>
          <td align="center" colspan="2">
            <strong>Changes of this bug report</strong>
          </td>
        </tr><table><tr>
        <td>
        <strong>Iulian&nbsp; edited on &nbsp;2005-11-16 15:09:18.0</strong>
        </td>
        </tr>
<tr><td bgcolor='#FFFFFF'>

</td></tr>
<tr>
        <td>
        <strong>Martin&nbsp; edited on &nbsp;2005-11-22 13:30:51.0</strong>
        </td>
        </tr>
<tr><td bgcolor='#FFFFFF'>
I assume it's fixed now? Sean, can you check?

 -- Martin
</td></tr></table>
<!--DisplayBugsBlock_End-->  </body>
</html>